src/components/expanded-state/AvailableNetworksv2.tsx

```diff
@@ -0,0 +1,229 @@
+      overflow: 'hidden',
+    },
+  };
+
+  const availableNetworks = useMemo(() => {
+    // we dont want to show mainnet
+    return Object.keys(networks)
+      .map(network => ethereumUtils.getNetworkFromChainId(Number(network)))
+      .filter(network => network !== Network.mainnet);
+  }, [networks]);
+
+  const { updateInputCurrency } = useSwapCurrencyHandlers({
+    shouldUpdate: true,
+    type: ExchangeModalTypes.swap,
+  });
+  const convertAssetAndNavigate = useCallback(
+    (chosenNetwork: Network) => {
+      const newAsset = asset;
+
+      // we need to convert the mainnet asset to the selected network's
+      newAsset.mainnet_address =
+        networks?.[ethereumUtils.getChainIdFromNetwork(Network.mainnet)]
+          ?.address ?? asset.address;
+      newAsset.address =
+        networks?.[ethereumUtils.getChainIdFromNetwork(chosenNetwork)].address;
+      newAsset.uniqueId = `${asset.address}_${chosenNetwork}`;
+      newAsset.type = chosenNetwork;
+
+      goBack();
+      navigate(Routes.EXCHANGE_MODAL, {
+        params: {
+          fromDiscover: true,
+          ignoreInitialTypeCheck: true,
+          defaultOutputAsset: newAsset,
+          type: CurrencySelectionTypes.input,
+          showCoinIcon: true,
+          title: lang.t('swap.modal_types.get_symbol_with', {
+            symbol: newAsset?.symbol,
+          }),
+          onSelectCurrency: updateInputCurrency,
+        },
+        screen: Routes.CURRENCY_SELECT_SCREEN,
+      });
+    },
+    [asset, goBack, navigate, networks, updateInputCurrency]
+  );
+
+  const handlePressContextMenu = useCallback(
+    ({ nativeEvent: { actionKey: network } }) => {
+      convertAssetAndNavigate(network);
+    },
+    [convertAssetAndNavigate]
+  );
+
+  const handlePressButton = useCallback(() => {
+    convertAssetAndNavigate(availableNetworks[0]);
+  }, [availableNetworks, convertAssetAndNavigate]);
+
+  const networkMenuItems = useMemo(() => {
+    return Object.values(networkInfo)
+      .filter(
+        ({ exchange_enabled, value }) =>
+          exchange_enabled &&
+          value !== Network.mainnet &&
+          !!networks[ethereumUtils.getChainIdFromNetwork(value)]
+      )
+      .map(netInfo => ({
+        actionKey: netInfo.value,
+        actionTitle: netInfo.name,
+        icon: {
+          iconType: 'ASSET',
+          iconValue: `${
+            netInfo.layer2 ? `${netInfo.value}BadgeNoShadow` : 'ethereumBadge'
+          }`,
+        },
+      }));
+  }, [networks]);
+
+  const MenuWrapper = availableNetworks.length > 1 ? ContextMenuButton : Box;
+
+  if (availableNetworks.length === 0) return null;
+  return (
+    <>
+      <MenuWrapper
+        // @ts-expect-error overloaded props ContextMenuButton
+        menuConfig={{ menuItems: networkMenuItems, menuTitle: '' }}
+        isMenuPrimaryAction
+        onPressMenuItem={handlePressContextMenu}
+        useActionSheetFallback={false}
+        style={{ paddingBottom: 27 }}
```

Comment:
this is inline styles since context menu isn't a DS component

Reply:
https://github.com/rainbow-me/rainbow/pull/4109#discussion_r961978964